src/graphql/admin/root/mutation/account-update-level.ts

```diff
@@ -0,0 +1,45 @@
+import { updateAccountLevel } from "@app/accounts/update-account-level"
+import AccountDetailPayload from "@graphql/admin/types/payload/account-detail"
+import AccountLevel from "@graphql/admin/types/scalar/account-level"
+import { mapError } from "@graphql/error-map"
+import { GT } from "@graphql/index"
+
+const AccountUpdateLevelInput = new GT.Input({
+  name: "AccountUpdateLevelInput",
+  fields: () => ({
+    uid: {
+      type: GT.NonNullID,
+    },
+    level: {
+      type: GT.NonNull(AccountLevel),
+    },
+  }),
+})
+
+const AccountUpdateLevelMutation = GT.Field({
+  type: GT.NonNull(AccountDetailPayload),
+  args: {
+    input: { type: GT.NonNull(AccountUpdateLevelInput) },
+  },
+  resolve: async (_, args) => {
+    const { uid, level } = args.input
+
+    for (const input of [uid, level]) {
+      if (input instanceof Error) {
+        return { errors: [{ message: input.message }] }
+      }
+    }
+
+    const account = await updateAccountLevel({ id: uid, level } as {
```

Comment:
Casting should happen in domain layer or via `checkedToXxx` function. Don't like this TODO - but not a blocker.